app/src/org/commcare/models/AsyncEntity.java

```diff
@@ -128,24 +124,14 @@ public String getNormalizedField(int i) {
 
     @Override
     public String getSortField(int i) {
-        //Get a db handle so we can get an outer lock
-        SQLiteDatabase db;
-        try {
-            db = CommCareApplication.instance().getUserDbHandle();
-        } catch (SessionUnavailableException e) {
-            return null;
-        }
-
-        //get the db lock
-        db.beginTransaction();
-        try {
+        if (EntityStorageCache.lockCache()) {

```

Comment:
What about making this implement `AutoCloseable` and then use `try-with-resource`?

Or perhaps `lockCache` returns an `AutoClosable` object and raises and exception if the transaction can't be started.